scripts/pylib/twister/twisterlib/testplan.py

```diff
@@ -78,6 +81,92 @@ class TestLevel:
     levels = []
     scenarios = []
 
+class Checker:
```

Comment:
personal opinion (not a request for change): I find this class and how it is handled hard to follow and it took me some time to understand what is going on. Especially when the tuples defined here are added to a `checks` list every time a given type of issue is found. I believe that using named fields would be easier to understand than using a logic based on a positional argument within a nested structure.